File: development/OnBoarding/OnBoarding/OnBoarding.Entities/ProjectPlan.cs
namespace OnBoarding.Entities
{
    using System;
    using System.Collections.Generic;

    public class ProjectPlan
    {
        public Guid Id { get; set; }

        public Guid ProjectId { get; set; }

        public virtual Project Project { get; set; }

       // public Guid RoleId { get; set; }

        // public virtual Role Role { get; set; }

        public int Week { get; set; }

        public int Day { get; set; }

        public Guid KnowledgeTransferId { get; set; }

        public virtual KnowledgeTransfer KnowledgeTransfer { get; set; }

        public virtual ICollection<AssociatePlan> AssociatePlan { get; set; } = new List<AssociatePlan>();

        public bool Active { get; set; }
    }
}

File: development/OnBoarding/OnBoarding/OnBoarding.Web.Fnz/src/models/ProjectPlan.ts
import { ProjectPlan, Project, PageRequest, KnowledgeTransfer } from '../typings/ApiClient';
import { Role } from '../typings/ApiClient';
import { IModalDialogProps } from './ModalDialog';

export interface IProjectPlanProps {
    loadProjectPlan(projectId?: string): void;
    loadProjectAndRole(): void;
    loadPageRequest(pageRequest?: PageRequest): void;
    loadProjectPlanProject: (projectId: string) => void;
    loadProjectPlanRole: (roleId: string) => void;
    loadAddProjectPlan(isAddEdit: boolean): void;
    projectPlan: ProjectPlan[];
    loadKnowledgeTransfer(projectId?: string): void;
    projectPlanDetails: ProjectPlanDetails;
    isAddEdit: boolean;
    Role: Role[];
    Project: Project[];
    KnowledgeTransfer: KnowledgeTransfer[];
    addProjectPlan: (projectPlan: ProjectPlan) => void;
    updateProjectPlan: (projectPlanDetails?: ProjectPlanDetails) => void;
    showModal(show: Boolean): void;
    dialogProps: IModalDialogProps;
    showDialog: Boolean;
    deleteProjectPlan: (id: string) => void;
    updateProjectPlanModel: (projectPlanModel?: ProjectPlanModel) => void;
    projectPlanModel: ProjectPlanModel;
    // OnPageChange(pageNumber: number): void;
}

export interface IProjectPlanState {
    ProjectPlan: ProjectPlan[];
    Role: Role[];
    Project: Project[];
    isAddEdit: boolean;
    KnowledgeTransfer: KnowledgeTransfer[];
    showDialog: Boolean;
    message: string;
    projectPlanModel: ProjectPlanModel;
    projectPlanDetails: ProjectPlanDetails;
}

export interface ProjectPlanDetails {
    id?: string;
    projectId?: string;
    roleId?: string;
    week?: number | undefined;
    day?: number | undefined;
    knowledgeTransferId?: string;
    selectedProject?: Project;
    selectedRole?: Role;
    role?: Role[];
    project?: Project[];
    projectPlan?: ProjectPlan[];
    knowledgeTransfer?: KnowledgeTransfer[];
    selectedProjectId?: string;
    selectedRoleId?: string;
}

export interface ProjectPlanModel {
    deletingProjectId?: string;
    
}